packages/lucia/src/middleware/index.ts

```diff
@@ -218,6 +218,36 @@ export const qwik = (): Middleware<[QwikRequestEvent]> => {
 	};
 };
 
+type ElysiaContext = {
+	request: Request
+	set: {
+		headers: Record<string, string>
+		status?: number | undefined
+		redirect?: string | undefined
+	}
+}
+export const elysia = (): Middleware<[ElysiaContext]> => {
+	return ({ args }) => {
+		const [{ request, set }] = args
+		const requestContext = {
+			request: {
+				url: request.url,
+				method: request.method,
+				headers: {
+					origin: request.headers.get('Origin'),
+					cookie: request.headers.get('Cookie'),
+					authorization: request.headers.get('Authorization'),
+				},
+			},
+			setCookie: (cookie: Cookie) => {
+				set.headers['Set-Cookie'] = set.headers['Set-Cookie'] ? `${set.headers['Set-Cookie']} ${cookie.serialize()}` : cookie.serialize()
```

Comment:
Looks like `Set-Cookie` can be an array? https://github.com/elysiajs/elysia-cookie/blob/99f609fac095566f71e762d831f368274cccee14/src/index.ts#L108

I wonder if we can get @SaltyAom to chime in

Reply:
the internal type would be lying https://github.com/elysiajs/elysia/blob/820b1e7e8c86ddc572e83d6d95b00443fc65ff68/src/context.ts#L19-L24